### ISSUE TITLE
case 0: {return;} errors

### ISSUE THREAD

**Issue by mgol** [eslint/eslint#893]
The following code:

``` js
/* global x: false */
switch (x) {
    case 0: { return; }
    case 1: { return; }
}
```

passes jsHint validation but errors with ESLint with:

```
_/test.js
  3:18  error  Illegal return statement
```
